### DISCUSSION TITLE
Autocompletions

### DISCUSSION THREAD

**Discussion by apastuszak** [Homebrew/discussions#3403] [Casks] [answered]
I've been running fish as my shell for a while now. Certain packages, when I installed them, tell me they've installed fish autocompletions.

A few days ago I switched to zsh and set up th zsh autocompletion plugin. I installed a package after that and it told me installed zsh autocompletions.

So, my question is, do I need to reinstall all my homebrew packages in order to get all the zsh autocompletions?

**Comment by carlocab** [comment, marked as answer]
> do I need to reinstall all my homebrew packages in order to get all the zsh autocompletions?

`brew` installs all completions wherever they're available, so you don't need to reinstall anything to use them. The only thing that changes when you use a different shell is the messages you seen upon installation.

**Reply by apastuszak** [reply]
Thank you for explaining that!

**Reply by carlocab** [reply]
Sure. Note that not all formulae provide completions, and some formulae provide completions only for some shells.

This means you might find that there are formulae you've installed that had completions for Fish but not Zsh (somewhat uncommon) or vice-versa (a little bit more common).
